hacking/target-clock-generation.md

```diff
@@ -0,0 +1,144 @@
+between TMS and TCK that we see in the capture.
+
+### Making the measurements
+
+We make multiple measurements in each of the following steps due to the issue of jitter caused by the capture
+temporal resolution resulting in quantisation noise. That is, if we measure only one cycle, the value measured
+may be off by a substantial enough factor that it introduces excessive error into subsequent calculations.
+We counter that by making several measurements of each section of pulses, and applying a standard average as this
+compensates for this capture jitter, improving the method's accuracy.
+
+![A detailed view of the first pulse section of interest](../_assets/target-clk-gen/first_pulse_train.png)
+
+We start by zooming in on the first pulse section of interest, and then using measurement markers to make
+several (at least 10) single-period measurements of the clock pulses generated by jtagtap_next(), making
+notes of the results and how many of each value are observed. Add the result up and divide by the total number
+of samples taken. Record the final result for later use.
+
+We then repeat for jtagtap_tms_seq(), making a record of the final result using the same multiple measurement
+and averaging method.
+
+![A detailed view of the second pulse section of interst](../_assets/target-clk-gen/second_pulse_train.png)
+
+Finally, we zoom back out, pan over and then zoomed back in on the second section of interest. We then repeat
+the measurement protocol on this final set of pulses and make a record of the final result after averaging.
+
+### Deducing the final TCK clock frequency
+
+After following the measurement protocol outlined above, you will now have 3 values - a clock frequency achieved
+by jtagtap_next(), jtagtap_tms_seq() and jtagtap_tdi_tdo_seq().
+
+It is tempting to take the fastest of these and say "that's the clock frequency!", but this would be misleading
+to the user and especially when trying to make comparative measurements to other probes and the clock frequencies
+they achieve. So instead, for comparative accuracy, we must average between these 3 measurements.
+
+Simply adding the 3 values up and dividng by 3 provides a reasonable approximation of the achieved clock frequency
+for BMP. With this done, using the workbook referenced in the introduction, note the final value in column
+A of the workbook against the delay factor line you have measured.
+
+### Additional measurements
+
+After following this guide to get a measurement of the _no_delay routines, which will act as a baseline for
+the highest achievable frequency output, we must then repeat with `target_clk_divider` set to the values
+0 through 4 (for 5 sets of additional measurements to the `_no_delay` one).
+
+These additional measurements provide the data required to perform a linear regression, and set the representation
+baseline when to apply the divider calculation.
```

Comment:
```suggestion
baseline when calculating the divider.
```

Reply:
Apologies for the confusing wording - it seems we missed a 'for' here. We were aiming to talk about the subsequent measurements providing the baseline (comparison point) for when the divider calculation should be applied. Please let us know if you still think this change should be applied or if this works in your mind:

```suggestion
baseline for when to apply the divider calculation.
```

Reply:
`for the divider calculation` would be a better alternative.